yb-voyager/cmd/exportSchema.go

```diff
@@ -325,49 +329,77 @@ func applySchemaTransformations() {
 		}
 	}
+
+	// Check the flag to message the user about the recommendations applied and ask to apply manually
+	if errorApplyingAssessmentRecommendations {
+		utils.PrintAndLog("\nUnable to apply assessment recommendations(sharded/colocated tables) to the exported schema. Please check the logs for more details.")
+		utils.PrintAndLog("You can apply the recommendations manually by referring to the assessment report.")
+	} else if assessmentRecommendationsApplied {
+		SetAssessmentRecommendationsApplied()
+	}
+	// else case will be whether neither applied nor errored, but rather schema file was not present.
+
+	// There is corner case: when recommmendations applied on table.sql but not on mview.sql or vice versa
+	// In this case, there is no definite answer whether assessmentRecommendationsApplied should be true or false; Assuming false.
 }
 
 // transformSchemaFile applies a sequence of transformations to the given schema file
 // and writes the transformed result back. If the file doesn't exist, logs a message and returns nil.
-func transformSchemaFile(filePath string, transformations []func(raw []*pg_query.RawStmt) ([]*pg_query.RawStmt, error), objectType string) error {
-	if !utils.FileOrFolderExists(filePath) {
-		log.Infof("%q file doesn't exist, skipping transformations for %s object type", filePath, objectType)
+func transformSchemaFile(filePath string, transformations []func(raw []*pg_query.RawStmt, filePath string) ([]*pg_query.RawStmt, error), objectType string) error {
+	if !utils.FileOrFolderExists(filePath) || len(transformations) == 0 {
+		log.Infof("schema file %q for object type %s doesn't exist or no transformations to apply", filePath, objectType)
 		return nil
 	}
 
-	rawStmts, err := queryparser.ParseSqlFile(filePath)
+	var rawStmts []*pg_query.RawStmt
+	var err error
+	defer func() {
+		if err != nil {
+			errorApplyingAssessmentRecommendations = true
+			utils.PrintAndLog("Failed to apply any transformation to the exported schema file %q: %v\n", filePath, err)
+		}
+	}()
+
+	log.Infof("applying transformations to the schema file %q for object type %s", filePath, objectType)
+	rawStmts, err = queryparser.ParseSqlFile(filePath)
 	if err != nil {
 		return fmt.Errorf("failed to parse sql statements from %s object type in schema file %q: %w", objectType, filePath, err)
 	}
 
-	beforeSqlStmts, err := queryparser.DeparseRawStmts(rawStmts)
+	var beforeSqlStmts []string
+	beforeSqlStmts, err = queryparser.DeparseRawStmts(rawStmts)
 	if err != nil {
 		return fmt.Errorf("failed to deparse raw stmts for %s object type in schema file %q: %w", objectType, filePath, err)
 	}
 
 	transformedStmts := rawStmts
 	// Apply transformations in order
 	for _, transformFn := range transformations {
-		newStmts, err := transformFn(transformedStmts)
-		if err != nil {
+		transformFuncName := utils.GetFuncName(transformFn)
+		log.Infof("applying transformation: %s on %s", filepath.Base(transformFuncName), filePath)
```

Comment:
I believe `filepath.Base` is for the `filePath` arg?

Reply:
transformFuncName also comes as fullpath to the function name